### DISCUSSION TITLE
Refactor quickbar using popper.js

### DISCUSSION THREAD

**Discussion by himself65** [toeverything/blocksuite#714] [General]
https://popper.js.org/

**Comment by doodlewind** [comment]
Or we may evolve our current implementation in QuickBar into an equivalent reusable library for positioning editor toolbars. What do you think about this @lawvs?

**Reply by himself65** [reply]
Anyway, I think the cost of maintaining a float positioning library is higher than using a popular tool

**Comment by lawvs** [comment]
We spend more time handling the direction of the selection instead of float positioning.

And popper.js cannot help me with this part.

But I will consider using it when developing other pop-up layers.

---

Specifically, the quick bar needs to display at end of the selection when the user selects some text. But the `getBoundingClientRect` is incorrect because the user's selection may not be a square

<img width="740" alt="Screenshot 2023-01-16 at 5 39 13 PM" src="https://user-images.githubusercontent.com/18554747/212647693-a605559d-e483-4644-b660-4f6ddd379753.png">

We handle it at 
https://github.com/toeverything/blocksuite/blob/49049693f2cd9e1daa3220b41e40d78aa2b4cb32/packages/blocks/src/page-block/utils/cursor.ts#L29-L140

And the popper.js can only do the following things

https://github.com/toeverything/blocksuite/blob/49049693f2cd9e1daa3220b41e40d78aa2b4cb32/packages/blocks/src/page-block/utils/cursor.ts#L142-L184
